app/src/main/java/com/xmartlabs/gong/domain/usecase/GetLocationUseCase.kt

```diff
@@ -2,15 +2,15 @@ package com.xmartlabs.gong.domain.usecase
 
 import com.xmartlabs.gong.data.model.Location
 import com.xmartlabs.gong.domain.repository.LocationRepository
-import com.xmartlabs.gong.domain.usecase.common.CoroutineUseCase
+import com.xmartlabs.gong.domain.usecase.common.FlowCoroutineUseCase
 
 /**
  * Created by mirland on 28/04/20.
  */
-interface GetLocationUseCase : CoroutineUseCase<GetLocationUseCase.Params, Location> {
+interface GetLocationUseCase : FlowCoroutineUseCase<GetLocationUseCase.Params, Location> {
   class Params
 }
 
 class GetLocationUseCaseImpl(private val locationRepository: LocationRepository) : GetLocationUseCase {
-  override suspend fun execute(params: GetLocationUseCase.Params): Location = locationRepository.getLocation()
+  override fun execute(params: GetLocationUseCase.Params) = locationRepository.getLocation()
```

Comment:
how does this work? has some delay? what happens when there is no data loaded?

Reply:
> has some delay? 
Why? It's the as you have a `Flowable<Location?>` if you are using rx (I know that you cannot have nullables in rx)
> what happens when there is no data loaded?
Returns `null`

Reply:
but it will have a delay?

Reply:
why? it's like a flowable, it's the same